### ISSUE TITLE
langchain_unstructured.UnstructuredLoader no longer supports loading document from a URL

### ISSUE THREAD

**Issue by sfitts** [langchain-ai/langchain#26230]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

The following works for the (now deprecated) `UnstructuredFileLoader` class:

```python
from langchain_community.document_loaders import UnstructuredFileLoader

loader = UnstructuredFileLoader(file_path=None, url="https://raw.githubusercontent.com/Unstructured-IO/unstructured-api/main/sample-docs/announcement.eml")
loader.load()
```

However, substituting the new `UnstructuredLoader` class:

```python
from langchain_unstructured import UnstructuredLoader

loader = UnstructuredLoader(file_path=None, url="https://raw.githubusercontent.com/Unstructured-IO/unstructured-api/main/sample-docs/announcement.eml")
loader.load()
```

### Description

I'm trying to load a document from its URL using `UnstructuredLoader`.  Doing this results in the error: "'Exactly one of file, filename and url must be specified."

The reason for this is that the `url` parameter is now interpreted as the URL for the unstructured API server and therefore is not present in `self.unstructured_kwargs`.  Thus by the time we end up calling the local "partition" method, it is impossible to pass it the `url` parameter that it supports.

There needs to be some way to pass this value through to the `partition` call.  IMO it would have been better to use something like `unstructured_api_url` as the parameter to set the URL for the unstructured API server, but that ship may have already sailed.  So now maybe there needs to be `document_url` or some such which can be added to `unstructured_kwargs` as `url`.  Another option is to only interpret `url` as the API server's URL when that option is enabled using `partition_via_api` (though that may be a bit too confusing).

For now, we are continuing to use the deprecated class since we need to be able to load documents via a URL.

### System Info

```
System Information
------------------
> OS:  Windows
> OS Version:  10.0.19045
> Python Version:  3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)]

Package Information
-------------------
> langchain_core: 0.2.38
> langchain: 0.2.16
> langchain_community: 0.2.16
> langsmith: 0.1.117
> langchain_aws: 0.1.17
> langchain_elasticsearch: 0.2.2
> langchain_google_genai: 1.0.10
> langchain_huggingface: 0.0.3
> langchain_nvidia_ai_endpoints: 0.2.2
> langchain_openai: 0.1.23
> langchain_qdrant: 0.1.1
> langchain_text_splitters: 0.2.4
> langchain_unstructured: 0.1.2

Optional packages not installed
-------------------------------
> langgraph
> langserve

Other Dependencies
------------------
> aiohttp: 3.10.5
> async-timeout: Installed. No version info available.
> boto3: 1.34.162
> dataclasses-json: 0.6.7
> elasticsearch[vectorstore-mmr]: Installed. No version info available.
> google-generativeai: 0.7.2
> httpx: 0.27.2
> huggingface-hub: 0.24.6
> jsonpatch: 1.33
> numpy: 1.26.4
> openai: 1.44.0
> orjson: 3.10.7
> packaging: 24.1
> pillow: 10.4.0
> pydantic: 1.10.18
> PyYAML: 6.0.2
> qdrant-client: 1.11.1
> requests: 2.32.3
> sentence-transformers: 3.0.1
> SQLAlchemy: 2.0.34
> tenacity: 8.5.0
> tiktoken: 0.7.0
> tokenizers: 0.19.1
> transformers: 4.44.2
> typing-extensions: 4.12.2
> unstructured-client: 0.24.1
> unstructured[all-docs]: Installed. No version info available.
```

**Comment by garychen2002** [comment]
Hello, I am from a group at UofT looking to contribute to langchain and am interested in solving this issue if you believe it is suitable to handle with the document_url approach you've mentioned.

**Comment by garychen2002** [comment]
It looks like this was solved in https://github.com/langchain-ai/langchain/commit/eef18dec442eabb2c2532bd67cc2efa12a43d406 unless I am mistaken.

**Comment by sfitts** [comment]
> It looks like this was solved in [eef18de](https://github.com/langchain-ai/langchain/commit/eef18dec442eabb2c2532bd67cc2efa12a43d406) unless I am mistaken.

It does seem that way -- not sure why this wasn't closed, but I'll confirm that it works and if so I'll close it manually.

**Comment by sfitts** [comment]
Confirmed as fixed.
